changelogs/fragments/801-fix-delete-without-ttl.yml

```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- route53 - ``ttl``  and ``value`` are not required for deleting records (https://github.com/ansible-collections/community.aws/pull/801).
```

Comment:
I think this is a breaking change. Until now, the behavior was that you must specify *exactly* the current DNS record set (including TTL and value), otherwise it won't get deleted. Now that's no longer required. Thus suddenly records will get deleted which weren't deleted before.

(I'm not saying that I don't like the new behavior better, I'm just saying that this will break existing playbooks and roles and potentially cause loss of data that users do not expect.)

Reply:
Sounds fixable via flags.. `no_ttl` `no_value` bool

Reply:
Yes, that's true.  

> Until now, the behavior was that you must specify exactly the current DNS record set (including TTL and value)

But on the other hand, the "exactly specification" is handled internally now.  
And furthermore, no one will use it like that in the wild, because until now, it throws a botocore exception.

Of course we can keep this PR for a later release and just introduce the diff mode fix with some breaking change notes for a future release. I'm also fine with that.

Reply:
> Sounds fixable via flags.. no_ttl no_value bool

I'm not a fan to introduce more parameters that will be need to drain out later also.  
[The current workaround is just to fetch the parameters of a record and keep them in the `absent` task.](https://github.com/konveyor/mig-agnosticd/issues/188#issuecomment-967283420)  
So the current situation is not a broken one imo :)


Let's fetch some more opinions.  
cc @tremble  @jillr  @alinabuzachis

Reply:
That 'workaround' has been the documented way of doing this for a looong time. See for example the examples in the 2.5 docs: https://docs.ansible.com/ansible/2.5/modules/route53_module.html (I'm sure that example has been there even longer, but the website doesn't seem to have older docs :) )

One idea would be to abuse the `overwrite` option: if `overwrite: true`, ignore TTL and value for deletion. That unfortunately would also be a breaking change though, since so far its value has been ignored. And someone might have specified `overwrite: true` together with `state: absent`...

Reply:
Suggestion - 3.0.0 is planned for Dec 6th, let's add it in the Dec 6 release.

Personally I think this is the right behaviour, TTL being mandatory during deletion is kinda weird, and adding different TTLs to the same RRSet is deprecated: https://www.rfc-editor.org/rfc/rfc2181#section-5.2